zkml/src/layers/mod.rs

```diff
@@ -75,7 +75,7 @@ pub enum Layer<T> {
     // TODO: so far it's only flattening the input tensor, e.g. new_shape = vec![shape.iter().product()]
     Flatten(Flatten),
     QKV(QKV<T>),
-    MhaQK(MhaQK),
+    MhaQK(Mha<T>),
```

Comment:
I won't fight hard on this but i find that keeping all the structs, smallest component possible in Layer is still worthwhile.  Or said differently, what's the reason to remove it ? 
It's like FFN, we dont do a FFN meta layer because we already have all the mini components tied up together.

Reply:
I think same answer as previous comment applies? :)